### PR TITLE
Feature/323 remove deprecations

### PR DESCRIPTION
**Pull request by umulmrum**
| Q             | A
| ------------- | ---
| Branch        | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | yes
| Deprecations? | no
| Tests pass?   | yes
| Fixed issues  | chameleon-system/chameleon-system#323
| License       | MIT

There are two code blocks commented out. While this is normally bad practice, I wanted to keep the information for reference. The Amazon Payment tests are broken and need to be redone (currently it looks like the bundle gets rewritten from scratch due to a new API), and there is no replacement for the commented-out code. So there is no "real" functional change, but developers can see how these tests were meant to work.

### CODE REVIEW

File: src/SearchBundle/objects/db/TShopSearchIndexer.class.php [hunk 1]
```diff
@@ -484,21 +484,7 @@ public static function GetIndexTableNameForIndexLength($iIndexLength)
      */
     public static function GetSearchQuery($sSearchTerm, $aSearchTerms = null, $aFilter = array(), $sLanguageId = null)
     {
-        if (false && '' === $sSearchTerm && null === $aSearchTerms && 0 === count($aFilter)) {
-            // so search term - so return query that selects nothing
```
Comment (UlrichKu) [inline src/SearchBundle/objects/db/TShopSearchIndexer.class.php]: many search! so select

File: src/SearchBundle/objects/db/TShopSearchKeywordArticleList.class.php [hunk 2]
```diff
@@ -22,8 +22,8 @@ class TShopSearchKeywordArticleList extends TAdbShopSearchKeywordArticleList
      */
     public static function &GetListForShopKeywords($iShopId, $aKeywordList, $iLanguageId = null)
     {
-        if (is_null($iLanguageId)) {
-            $iLanguageId = TGlobal::GetActiveLanguageId();
+        if ($iLanguageId === null) {
```
Comment (UlrichKu) [inline src/SearchBundle/objects/db/TShopSearchKeywordArticleList.class.php]: null ===

also below

File: src/SearchBundle/objects/db/TShopSearchIndexer.class.php (hunk 1 above)
Comment (umulmrum) [reply src/SearchBundle/objects/db/TShopSearchIndexer.class.php]: wow!

File: src/SearchBundle/objects/db/TShopSearchKeywordArticleList.class.php (hunk 2 above)
Comment (umulmrum) [reply src/SearchBundle/objects/db/TShopSearchKeywordArticleList.class.php]: Done.

**Comment by umulmrum** [comment]
Added one commit with loads of additional changes.

**Pull request merged**